ParserTests/CommentsTests.cs

```diff
@@ -235,4 +461,618 @@ public void TestMixedEOLComment()
+
+            Assert.Equal(SingleLineCommentsToken.INT, token1.TokenID);
+            Assert.Equal("1", token1.Value);
+            Assert.Equal(0, token1.Position.Line);
+            Assert.Equal(0, token1.Position.Column);
+            Assert.Equal(SingleLineCommentsToken.INT, token2.TokenID);
+            Assert.Equal("2", token2.Value);
+            Assert.Equal(1, token2.Position.Line);
+            Assert.Equal(0, token2.Position.Column);
+            Assert.Equal(SingleLineCommentsToken.COMMENT, token3.TokenID);
+            Assert.Equal(" single line comment", token3.Value.Replace("\r", "").Replace("\n", ""));
+            Assert.Equal(1, token3.Position.Line);
+            Assert.Equal(2, token3.Position.Column);
+            Assert.Equal(SingleLineCommentsToken.DOUBLE, token4.TokenID);
+            Assert.Equal("3.0", token4.Value);
+            Assert.Equal(2, token4.Position.Line);
+            Assert.Equal(0, token4.Position.Column);
+        }
+
+        [Fact]
+        public void TestInnerMultiComment()
+        {
+            var lexerRes = LexerBuilder.BuildLexer(new BuildResult<ILexer<SingleLineCommentsToken>>());
+            Assert.False(lexerRes.IsError);
+            var lexer = lexerRes.Result as GenericLexer<SingleLineCommentsToken>;
+
+            var dump = lexer.ToString();
+
+            var code = @"1
+2 /* inner */ 3
+4
+            ";
+
+            var tokens = lexer.Tokenize(code).ToList();
+
+            Assert.Equal(5, tokens.Count);
+
+            var token1 = tokens[0];
+            var token2 = tokens[1];
+            var token3 = tokens[2];
+            var token4 = tokens[3];
+
+            Assert.Equal(SingleLineCommentsToken.INT, token1.TokenID);
+            Assert.Equal("1", token1.Value);
+            Assert.Equal(0, token1.Position.Line);
+            Assert.Equal(0, token1.Position.Column);
+
+            Assert.Equal(SingleLineCommentsToken.INT, token2.TokenID);
+            Assert.Equal("2", token2.Value);
+            Assert.Equal(1, token2.Position.Line);
+            Assert.Equal(0, token2.Position.Column);
+
+            Assert.Equal(SingleLineCommentsToken.INT, token3.TokenID);
+            Assert.Equal("3", token3.Value);
+            Assert.Equal(1, token3.Position.Line);
+            Assert.Equal(14, token3.Position.Column);
+
+            Assert.Equal(SingleLineCommentsToken.INT, token4.TokenID);
+            Assert.Equal("4", token4.Value);
+            Assert.Equal(2, token4.Position.Line);
+            Assert.Equal(0, token4.Position.Column);
+        }
+
+        [Fact]
+        public void TestMixedEOLComment()
+        {
+            var lexerRes = LexerBuilder.BuildLexer(new BuildResult<ILexer<SingleLineCommentsToken>>());
+            Assert.False(lexerRes.IsError);
+            var lexer = lexerRes.Result as GenericLexer<SingleLineCommentsToken>;
+
+            var dump = lexer.ToString();
+            var code = "1\n2\r\n/* multi line \rcomment on 2 lines */ 3.0";
```

Comment:
How should the parser behave here? Right now it ignores the comment as well as the DOUBLE (3.0), generating two tokens for the INT and one empty token at the end.

I would expect the comments and text to be completely ignored and the INT 2 and DOUBLE 3.0 be tokenized or to completely fail compiling with an error. The actual behavior seems erroneous for me.

Reply:
Still the same. To states clearly : 

1. If only SingleLineComment attribute, then: 
    - single line comment allowed
    - multi line comment forbidden
2. if only MultiLigneComment attribute then:
   - single line comment forbidenn
  - mult line comment allowed
3. if both MultiLineComment AND SingleLineComment then both single line and multiline comments are allowed 
4. if Comment(x) or Comment(x,nullorempty,nullorempty) : same as case 1
- if Comment(x,y,z) same as case 3

And I would prefer that mixing CommentAttribute and Single/MultiLineAttribute would not be allowed for clarity.

Hope I do not miss some possible combination

Does it seems ok for you

Reply:
Out of item 4, these points are already on the unit test cases, with negative and positive tests.

> Still the same. To states clearly :
> 
> 1. If only SingleLineComment attribute, then:
>    
>    * single line comment allowed
>    * multi line comment forbidden
> 2. if only MultiLigneComment attribute then:
>    
>    * single line comment forbidenn
> 
> * mult line comment allowed
> 
> 1. if both MultiLineComment AND SingleLineComment then both single line and multiline comments are allowed
> 2. if Comment(x) or Comment(x,nullorempty,nullorempty) : same as case 1
> 
> * if Comment(x,y,z) same as case 3
> 
> And I would prefer that mixing CommentAttribute and Single/MultiLineAttribute would not be allowed for clarity.
> 
> Hope I do not miss some possible combination
> 
> Does it seems ok for you